ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala

```diff
@@ -0,0 +1,182 @@
+          exerciseResult = None,
+          key = None,
+        )
+      )
+      builder.add(
+        NodeFetch(
+          coid = create2,
+          templateId = someTemplateId,
+          optLocation = None,
+          actingParties = Some(Set(bob)),
+          signatories = Set(bob),
+          stakeholders = Set(bob),
+          key = Some(
+            KeyWithMaintainers(ValueParty(bob), Set(bob))
+          ),
+        ),
+        parent = rootExercise,
+      )
+      val nestedExercise = builder.add(
+        NodeExercises(
+          targetCoid = create2,
+          templateId = someTemplateId,
+          choiceId = Ref.ChoiceName.assertFromString("SomeChoice"),
+          optLocation = None,
+          consuming = true,
+          actingParties = Set(bob),
+          chosenValue = someValueRecord,
+          stakeholders = Set(bob),
+          signatories = Set(bob),
+          children = ImmArray.empty,
+          exerciseResult = None,
+          key = Some(
+            KeyWithMaintainers(ValueParty(bob), Set(bob))
+          ),
+        )
+      )
+      builder.add(
+        NodeCreate(
+          coid = builder.newCid,
+          coinst = someContractInstance,
+          optLocation = None,
+          signatories = Set(bob),
+          stakeholders = Set(alice, bob),
+          key = Some(
+            KeyWithMaintainers(ValueParty(bob), Set(bob))
+          )
+        ),
+        parent = nestedExercise,
+      )
+      Tx.CommittedTransaction(builder.build())
+    }
+
+    val someVersionedContractInstance =
+      ContractInst(
+        template = someContractInstance.template,
+        agreementText = someContractInstance.agreementText,
+        arg = VersionedValue(
+          version = ValueVersion("6"),
+          value = someContractInstance.arg
+        )
+      )
+
+    val t1 = Instant.now()
+    val t2 = t1.plusMillis(1)
+    val t3 = t2.plusMillis(1)
+    val appId = UUID.randomUUID.toString
+    for {
+      _ <- store(
+        nextOffset() -> LedgerEntry.Transaction(
+          commandId = Some(UUID.randomUUID.toString),
+          transactionId = UUID.randomUUID.toString,
+          applicationId = Some(appId),
+          submittingParty = Some(alice),
+          workflowId = None,
+          ledgerEffectiveTime = t1,
+          recordedAt = t1,
+          transaction = tx1,
+          explicitDisclosure = Map.empty,
+        )
+      )
+      _ <- store(
+        nextOffset() -> LedgerEntry.Transaction(
+          commandId = Some(UUID.randomUUID.toString),
+          transactionId = UUID.randomUUID.toString,
+          applicationId = Some(appId),
+          submittingParty = Some(bob),
+          workflowId = None,
+          ledgerEffectiveTime = t2,
+          recordedAt = t2,
+          transaction = tx2,
+          explicitDisclosure = Map.empty,
+        )
+      )
+      _ <- store(
+        divulgedContracts = Map((create1, someVersionedContractInstance) -> Set(alice)),
```

Comment:
The fetch node that is doing the divulgence fetches `create2`. Should this be used here as well?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6607/commits/070f7dd7949de123f813e57eec30911c7f400f3e